File: test-api.py
#!/usr/bin/env python3

import requests
import json
import sys

BASE_URL = "https://50-deedscom-enterprise-db0653f4.base44.app/api/functions/enterpriseApi"
API_KEY = "0a5ae0c87fe64466b37092a7d42acd77e1da29e8c50e607991da5b0b8d5a6718"

HEADERS = {
    "Content-Type": "application/json",
}

def log_request(method, path, body=None):
    """Log the request details."""
    print(f"\n{'='*70}")
    print(f"REQUEST: {method} {path}")
    print(f"URL: {BASE_URL}")
    print(f"Headers: Content-Type: application/json")
    if body:
        print(f"Body (with _api_key): {json.dumps(body, indent=2)}")
    print('='*70)

def log_response(response):
    """Log the response details."""
    print(f"\nRESPONSE Status: {response.status_code}")
    print(f"Headers: {dict(response.headers)}")
    try:
        parsed = response.json()
        print(f"Body: {json.dumps(parsed, indent=2)}")
    except:
        print(f"Body: {response.text}")

def check_status(response, expected_status, test_name):
    """Check if response status matches expected."""
    passed = response.status_code == expected_status
    status = "✓ PASS" if passed else "✗ FAIL"
    print(f"\n{status}: {test_name}")
    print(f"  Expected: {expected_status}, Got: {response.status_code}")
    return passed

def test_get_pricing():
    """Test: Get Pricing (FL/Walton - has specific pricing)"""
    api_path = "/pricing/FL/Walton"
    log_request("GET", api_path)
    try:
        payload = {
            "_path": api_path,
            "_method": "GET",
            "_api_key": API_KEY,
            "deed_type": "Transfer to Individual: FinCEN non-reportable",
        }
        response = requests.post(BASE_URL, json=payload, headers=HEADERS, timeout=10)
        log_response(response)
        # Should return 200 with Walton pricing
        return check_status(response, 200, "Get Pricing - Should return 200 with Walton pricing")
    except requests.exceptions.RequestException as e:
        print(f"✗ FAIL: Get Pricing - Request failed")
        print(f"Error: {str(e)}")
        return False

def test_list_orders():
    """Test: List Orders"""
    api_path = "/orders"
    log_request("GET", api_path)
    try:
        payload = {
            "_path": api_path,
            "_method": "GET",
            "_api_key": API_KEY,
        }
        response = requests.post(BASE_URL, json=payload, headers=HEADERS, timeout=10)
        log_response(response)
        return check_status(response, 200, "List Orders - Should return 200")
    except requests.exceptions.RequestException as e:
        print(f"✗ FAIL: List Orders - Request failed")
        print(f"Error: {str(e)}")
        return False

def test_get_specific_order(order_id=None):
    """Test: Get Specific Order"""
    api_path = f"/orders/{order_id}" if order_id else "/orders/invalid-id"
    log_request("GET", api_path)
    try:
        payload = {
            "_path": api_path,
            "_method": "GET",
            "_api_key": API_KEY,
        }
        response = requests.post(BASE_URL, json=payload, headers=HEADERS, timeout=10)
        log_response(response)
        return check_status(response, 200, "Get Specific Order - Should return 200")
    except requests.exceptions.RequestException as e:
        print(f"✗ FAIL: Get Specific Order - Request failed")
        print(f"Error: {str(e)}")
        return False

def test_create_order():
    """Test: Create Order"""
    api_path = "/orders"
    order_data = {
        "property_address": "123 Main St, Miami, FL 33101",
        "grantor_name": "John Doe",
        "grantee_name": "Jane Doe",
        "contact_name": "John Doe",
        "deed_type": "Transfer from entity to Trust: FinCEN reportable",
        "county": "Miami-Dade",
        "state": "FL",
        "contact_email": "test@example.com",
        "additional_instructions": "Test order",
        "attachments": [
            {
                "file_url": "https://example.com/documents/deed_draft.pdf",
                "file_name": "deed_draft.pdf",
                "file_size": 102400,
            }
        ],
    }
    log_request("POST", api_path, order_data)
    try:
        payload = {
            "_path": api_path,
            "_method": "POST",
            "_api_key": API_KEY,
            **order_data,
        }
        response = requests.post(BASE_URL, json=payload, headers=HEADERS, timeout=10)
        log_response(response)
        return check_status(response, 201, "Create Order - Should return 201")
    except requests.exceptions.RequestException as e:
        print(f"✗ FAIL: Create Order - Request failed")
        print(f"Error: {str(e)}")
        return False

def test_register_webhook():
    """Test: Register Webhook"""
    api_path = "/webhooks/register"
    webhook_data = {
        "url": "https://your-endpoint.com/webhook",
    }
    log_request("POST", api_path, webhook_data)
    try:
        payload = {
            "_path": api_path,
            "_method": "POST",
            "_api_key": API_KEY,
            **webhook_data,
        }
        response = requests.post(BASE_URL, json=payload, headers=HEADERS, timeout=10)
        log_response(response)
        return check_status(response, 201, "Register Webhook - Should return 201")
    except requests.exceptions.RequestException as e:
        print(f"✗ FAIL: Register Webhook - Request failed")
        print(f"Error: {str(e)}")
        return False

def main():
    """Run all tests."""
    print(f"\n{'█'*70}")
    print("Enterprise API Test Suite")
    print(f"{'█'*70}")

    results = []
    order_id = None

    # Run tests
    results.append(("Get Pricing (FL/Miami-Dade)", test_get_pricing()))

    # Test List Orders and capture an order ID
    list_orders_passed = test_list_orders()
    results.append(("List Orders", list_orders_passed))

    # Try to get an order ID for the next test
    try:
        payload = {
            "_path": "/orders",
            "_method": "GET",
            "_api_key": API_KEY,
        }
        response = requests.post(BASE_URL, json=payload, headers=HEADERS, timeout=10)
        if response.status_code == 200:
            data = response.json()
            if data.get("orders") and len(data["orders"]) > 0:
                order_id = data["orders"][0]["id"]
    except:
        pass

    results.append(("Get Specific Order", test_get_specific_order(order_id)))
    results.append(("Create Order", test_create_order()))
    results.append(("Register Webhook", test_register_webhook()))

    # Summary
    print(f"\n{'█'*70}")
    print("TEST SUMMARY")
    print(f"{'█'*70}")
    for test_name, passed in results:
        status = "✓" if passed else "✗"
        print(f"{status} {test_name}")

    passed_count = sum(1 for _, passed in results if passed)
    total_count = len(results)
    print(f"\nTotal: {passed_count}/{total_count} passed")
    print(f"{'█'*70}\n")

    sys.exit(0 if passed_count == total_count else 1)

if __name__ == "__main__":
    main()

File: test-api.js
#!/usr/bin/env node

const https = require('https');

const BASE_URL = 'https://50-deedscom-enterprise-db0653f4.base44.app/api/functions/enterpriseApi';
const API_KEY = 'fc779b2e4c79cecec9f995d5098eac8ae8ba4e6ccd289ea9cf9ce3b8fbd95261';

// Utility function to make HTTPS requests
// Base44 functions take the path as a request body parameter, not URL path
function makeRequest(method, apiPath, body = null) {
  return new Promise((resolve, reject) => {
    const url = new URL(BASE_URL);

    // Construct the request body with path and method
    const requestBody = {
      path: apiPath,
      method: method,
      ...(body || {}),
    };

    const options = {
      hostname: url.hostname,
      path: url.pathname,
      method: 'POST', // Base44 functions always receive POST
      headers: {
        'Authorization': `Bearer ${API_KEY}`,
        'Content-Type': 'application/json',
      },
    };

    console.log(`\n${'='.repeat(70)}`);
    console.log(`REQUEST: ${method} ${apiPath}`);
    console.log(`Base44 Function URL: ${BASE_URL}`);
    console.log(`Headers: Authorization: Bearer ${API_KEY.substring(0, 20)}...`);
    console.log(`Body:`);
    console.log(`${JSON.stringify(requestBody, null, 2)}`);
    console.log('='.repeat(70));

    const req = https.request(options, (res) => {
      let data = '';

      res.on('data', (chunk) => {
        data += chunk;
      });

      res.on('end', () => {
        console.log(`\nRESPONSE Status: ${res.statusCode}`);
        try {
          const parsed = JSON.parse(data);
          console.log(`Body: ${JSON.stringify(parsed, null, 2)}`);
        } catch (e) {
          console.log(`Body: ${data}`);
        }

        resolve({
          status: res.statusCode,
          headers: res.headers,
          body: data,
          parsedBody: tryParseJSON(data),
        });
      });
    });

    req.on('error', reject);

    // Write the request body with path and method
    req.write(JSON.stringify(requestBody));
    req.end();
  });
}

function tryParseJSON(str) {
  try {
    return JSON.parse(str);
  } catch (e) {
    return str;
  }
}

function checkStatus(response, expectedStatus, testName) {
  const pass = response.status === expectedStatus;
  const status = pass ? '✓ PASS' : '✗ FAIL';
  console.log(`\n${status}: ${testName}`);
  console.log(`  Expected: ${expectedStatus}, Got: ${response.status}`);
  return pass;
}

async function runTests() {
  const results = [];

  console.log('\n' + '█'.repeat(70));
  console.log('Enterprise API Test Suite');
  console.log('█'.repeat(70));

  // Test 1: Get Pricing
  try {
    const pricingResponse = await makeRequest('GET', '/pricing/FL/Miami-Dade');
    results.push({
      name: 'Get Pricing (FL/Miami-Dade)',
      pass: checkStatus(pricingResponse, 200, 'Get Pricing - Should return 200'),
    });
  } catch (error) {
    console.error('✗ FAIL: Get Pricing - Request failed');
    console.error(`Error: ${error.message}`);
    results.push({ name: 'Get Pricing (FL/Miami-Dade)', pass: false });
  }

  // Test 2: List Orders
  try {
    const ordersResponse = await makeRequest('GET', '/orders');
    results.push({
      name: 'List Orders',
      pass: checkStatus(ordersResponse, 200, 'List Orders - Should return 200'),
    });
  } catch (error) {
    console.error('✗ FAIL: List Orders - Request failed');
    console.error(`Error: ${error.message}`);
    results.push({ name: 'List Orders', pass: false });
  }

  // Test 3: Get Specific Order (using a test ID)
  try {
    const orderResponse = await makeRequest('GET', '/orders/order123');
    results.push({
      name: 'Get Specific Order',
      pass: checkStatus(orderResponse, 200, 'Get Specific Order - Should return 200'),
    });
  } catch (error) {
    console.error('✗ FAIL: Get Specific Order - Request failed');
    console.error(`Error: ${error.message}`);
    results.push({ name: 'Get Specific Order', pass: false });
  }

  // Test 4: Create Order
  const createOrderPayload = {
    property_address: '123 Main St, Miami, FL 33101',
    grantor_name: 'John Doe',
    deed_type: 'Quitclaim Deed between Individuals (Add/Remove name)',
    county: 'Miami-Dade',
    state: 'FL',
    additional_instructions: 'Test order',
  };

  try {
    const createResponse = await makeRequest('POST', '/orders', createOrderPayload);
    const isExpectedStatus = createResponse.status === 201 || createResponse.status === 200;
    results.push({
      name: 'Create Order',
      pass: checkStatus(createResponse, 201, 'Create Order - Should return 201'),
    });
  } catch (error) {
    console.error('✗ FAIL: Create Order - Request failed');
    console.error(`Error: ${error.message}`);
    results.push({ name: 'Create Order', pass: false });
  }

  // Test 5: Register Webhook
  const webhookPayload = {
    url: 'https://your-endpoint.com/webhook',
  };

  try {
    const webhookResponse = await makeRequest('POST', '/webhooks/register', webhookPayload);
    const isExpectedStatus = webhookResponse.status === 200 || webhookResponse.status === 201;
    results.push({
      name: 'Register Webhook',
      pass: checkStatus(webhookResponse, 200, 'Register Webhook - Should return 200 or 201'),
    });
  } catch (error) {
    console.error('✗ FAIL: Register Webhook - Request failed');
    console.error(`Error: ${error.message}`);
    results.push({ name: 'Register Webhook', pass: false });
  }

  // Summary
  console.log('\n' + '█'.repeat(70));
  console.log('TEST SUMMARY');
  console.log('█'.repeat(70));
  results.forEach((result) => {
    const status = result.pass ? '✓' : '✗';
    console.log(`${status} ${result.name}`);
  });

  const passCount = results.filter((r) => r.pass).length;
  const totalCount = results.length;
  console.log(`\nTotal: ${passCount}/${totalCount} passed`);
  console.log('█'.repeat(70) + '\n');

  process.exit(passCount === totalCount ? 0 : 1);
}

runTests().catch((error) => {
  console.error('Fatal error:', error);
  process.exit(1);
});
